VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/promotion-expressions.js

```diff
@@ -22,52 +22,165 @@ angular.module('virtoCommerce.marketingModule')
                 },
                 canExecuteMethod: function () {
-                    return selectedListEntries.length == 1;
+                    return true;
                 }
-            }]
-        };
+            },
+            {
+                name: "platform.commands.delete", icon: 'fa fa-trash-o',
+                executeMethod: function (blade) {
+                    var grid = blade.$scope.gridApi.grid;
+                    var selectedRows = grid.api.selection.getSelectedRows();
+                    for (var entry of selectedRows) {
+                        grid.options.data = _.filter(grid.options.data, function (item) {
+                            return item.id != entry.id;
+                        });
+                    }
 
-        newBlade.options = {
-            showCheckingMultiple: false,
-            checkItemFn: function (listItem, isSelected) {
-                if (listItem.type == 'category') {
-                    newBlade.error = 'Must select Product';
-                    listItem.selected = undefined;
-                } else {
-                    if (isSelected) {
-                        if (_.all(selectedListEntries, function (x) { return x.id != listItem.id; })) {
-                            selectedListEntries.push(listItem);
-                        }
+                    blade.selectedListEntries = grid.options.data.slice();
+                    parentElement.productIds = _.map(grid.options.data, function (item) {
+                        return item.id;
+                    });
+                    blade.productIds = parentElement.productIds.slice();
+                },
+                canExecuteMethod: function (blade) {
+                    if (blade.childrenBlades.length > 0) {
+                        return false;
                     }
-                    else {
-                        selectedListEntries = _.reject(selectedListEntries, function (x) { return x.id == listItem.id; });
+                    if (blade.$scope.gridApi) {
+                        return blade.$scope.gridApi.grid.api.selection.getSelectedRows().length > 0;
                     }
-                    newBlade.error = undefined;
+                    return false;
                 }
-            }
+            }]
+        };
+
+        newBlade.refresh = function () {
+            newBlade.isLoading = true;
+            items.query({ ids: newBlade.productIds, respGroup: 'ItemInfo' }, function (data) {
+                newBlade.$scope.gridApi.grid.options.data = data;
+                newBlade.selectedListEntries = newBlade.$scope.gridApi.grid.options.data.slice();
+                newBlade.isLoading = false;
+            });
         };
 
+        newBlade.productIds = parentElement.productIds != undefined ? parentElement.productIds : [];
+        newBlade.selectedListEntries = selectedListEntries;
         bladeNavigationService.showBlade(newBlade, $scope.blade);
-    }
+    };
```

Comment:
Is is needed?

Reply:
Yes, this is assignment, not just a function definition.